Source/quests.cpp

```diff
@@ -53,96 +53,65 @@ int QuestGroup4[2] = { QTYPE_VEIL, QTYPE_WARLRD };
 
 void __cdecl InitQuests()
 {
-	char v0;           // dl
-	unsigned char *v1; // esi
-	unsigned char *v2; // eax
-	unsigned char *v3; // ecx
-	int *v4;           // eax
-	int v5;            // ebp
-	unsigned int v6;   // edi
-	//int v7; // eax
-	unsigned char v8; // al
-	unsigned char v9; // al
-	char v10;         // al
-	int v13;          // eax
-	int v15;          // eax
-	int v17;          // eax
-	int v19;          // eax
-	char v20;         // [esp+8h] [ebp-4h]
-
-	v0 = gbMaxPlayers;
-	v1 = &quests[0]._qactive;
+	int initiatedQuests;
+	int i;
+	
 	if (gbMaxPlayers == 1) {
-		v2 = &quests[0]._qactive;
-		do {
-			*v2 = 0;
-			v2 += 24;
-		} while ((signed int)v2 < (signed int)&quests[MAXQUESTS]._qactive);
+		for (i = 0; i < MAXQUESTS; i++) {
+			quests[i]._qactive = 0;
+		}
 	} else {
-		v3 = &quests[0]._qactive;
-		v4 = &questlist[0]._qflags;
-		do {
-			if (!(*(_BYTE *)v4 & 1))
-				*v3 = 0;
-			v4 += 5;
-			v3 += 24;
-		} while ((signed int)v4 < (signed int)&questlist[MAXQUESTS]._qflags);
+		for (i = 0; i < MAXQUESTS; i++) {
+			if (!(questlist[i]._qflags & 1)) {
+				quests[i]._qactive = 0;
+			}
+		}
 	}
-	v5 = 0;
+
+	initiatedQuests = 0;
 	questlog = 0;
 	ALLQUESTS = 1;
 	WaterDone = 0;
-	v20 = 0;
-	v6 = 0;
-	do {
-		if ((unsigned char)v0 <= 1u || questlist[v6]._qflags & 1) {
-			*(v1 - 1) = questlist[v6]._qdtype;
-			if ((unsigned char)v0 <= 1u) {
-				v8 = questlist[v6]._qdlvl;
-				*v1 = 1;
-				*(v1 - 2) = v8;
-				v1[13] = 0;
-				*(_DWORD *)(v1 + 18) = 0;
-			} else {
-				*(v1 - 2) = questlist[v6]._qdmultlvl;
-				//_LOBYTE(v7) = delta_quest_inited(v5);
-				if (!delta_quest_inited(v5)) {
-					*v1 = 1;
-					v1[13] = 0;
-					*(_DWORD *)(v1 + 18) = 0;
+
+	for (i = 0; i < MAXQUESTS; i++) {

```

Comment:
The counter in this loop is different than the one from the two previous since it is unsigned and they are signed.

Reply:
Good catch I completely missed that, the jb -jl difference is now gone.